crates/composer/src/main.rs

```diff
@@ -0,0 +1,37 @@
+use composer::api::Event;
+use composer::api::DEFAULT_SERVER_ADDRESS;
+use eyre::Result;
+use rodio::{source::Source, Decoder, OutputStream};
+use std::fs::File;
+use std::io::BufReader;
+use std::net::UdpSocket;
+
+fn main() -> Result<()> {
+    color_eyre::install()?;
+
+    let socket = UdpSocket::bind(DEFAULT_SERVER_ADDRESS)?;
+    println!("Listening on {}", socket.local_addr()?);
+
+    let (_stream, stream_handle) = OutputStream::try_default()?;
+
+    loop {
+        if let Err(err) = {
+            // Should be more than enough.
+            let mut buf = [0; 1000];
+            let (number_of_bytes, _) = socket.recv_from(&mut buf)?;
+
+            let event: Event = bincode::deserialize(&buf)?;
+            println!("Received an event ({number_of_bytes} bytes): {:?}", event);
+
+            // FIXME: do the decoding and file reading outside the loop
+            let file = BufReader::new(File::open("src/sound_samples/click.wav")?);
+            let source = Decoder::new(file)?;
+            stream_handle.play_raw(source.convert_samples())?;
+
+            Ok::<(), eyre::Error>(())
```

Comment:
Does it not compile without `::<(), eyre::Error>`?

https://docs.rs/eyre/latest/eyre/type.Result.html#example

Reply:
See my comment on the control flow issue; this is indeed needed as it's being passed to the `if let` expression at the top, but it's likely not what we want.

Reply:
Resolved as part of 5a375640f9e1e471bf9fc2fe618e8672dfdc148c